pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -1188,3 +1188,90 @@ func TestFairEvaluationForNodes(t *testing.T) {
 		}
 	}
 }
+
+func TestFitErrorNoSpammy(t *testing.T) {
+	tests := []struct {
+		name           string
+		nodeNum1       int
+		nodeNum2       int
+		addFalseFilter bool
+		wErr           string
+	}{
+		{
+			name:     "small number of nodes cannot pass the filters",
+			nodeNum1: 2,
+			wErr:     "0/2 nodes are available: 2 injecting failure for pod fake.",
+		},
+		{
+			name:     "large number of nodes cannot pass the filters",
+			nodeNum1: 500,
+			wErr:     "0/500 nodes are available: 500 injecting failure for pod fake.",
+		},
+		{
+			name:           "cannot pass the filters with two different reasons",
+			nodeNum1:       490,
+			nodeNum2:       10,
+			addFalseFilter: true,
+			wErr:           "0/500 nodes are available: 10 Nodes failed the fake plugin, 490 injecting failure for pod fake.",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			pod := &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "fake", UID: types.UID("fake")}}
+			cache := internalcache.New(time.Duration(0), wait.NeverStop)
+			var nodes []*v1.Node
+			failedNodeReturnCodeMap := make(map[string]framework.Code)
+			for i := 0; i < test.nodeNum1; i++ {
+				uid := fmt.Sprintf("node%v", i)
+				node := &v1.Node{ObjectMeta: metav1.ObjectMeta{Name: uid, UID: types.UID(uid)}}
+				nodes = append(nodes, node)
+				cache.AddNode(node)
+				failedNodeReturnCodeMap[uid] = framework.Unschedulable
```

Comment:
What was the map for? shouldn't this be `framework.Error`?

Reply:
it's mimic `Unschedulable` status and `feasibleNodes` is `0`, so that `FitError` could be returned.

Reply:
if it's not `unschedulable` status, it will not return `FitError` ,

https://github.com/kubernetes/kubernetes/blob/4f9317596ccb9fcb37bf183babf3e8640afcca9b/pkg/scheduler/core/generic_scheduler.go#L111-L122

and then the error message won't roll up to one line by below method, 
```func (f *FitError) Error()```